java/src/test/java/ai/rapids/cudf/BinaryOpTest.java

```diff
@@ -283,6 +315,31 @@ public void testAdd() {
         assertColumnsAreEqual(addIntFirst, addDoubleFirst, "int + double vs double + int");
       }
 
+      try (ColumnVector add = dec32cv1.add(dec32cv2)) {
```

Comment:
Isn't this is setting the result scale of the result to 0?  If so how is it that this test passes? or is the underlying cudf code ignoring out output type?

Reply:
Yes, you are right :). The underlying cudf code is ignoring out output type and creating  a new one(After calculating the new scale which depends on the operation). 
For fixed point binary operation, output_type isn't passed - https://github.com/rapidsai/cudf/blob/branch-0.17/cpp/src/binaryop/binaryop.cpp#L699  and instead created here: https://github.com/rapidsai/cudf/blob/branch-0.17/cpp/src/binaryop/binaryop.cpp#L579
For the same reason I was creating a DType with `scale=0` above in BinaryOperable.java as that would not be considered in cudf code.

Reply:
Then we need to document it and let users of the APIs know that if they do pass in a type it is ignored, or in exactly which situations it is ignored.  For example, If we passed in a FLOAT64 as the output would it be ignored or would it be an error?

Reply:
The output type would be ignored if we pass in any type such as FLOAT64 in this case. The operation would be performed making DECIMAL type as output type. I will update the documentation for all the API's(add, sub... ) which ignores the output type for decimal type operations.